src/libbson/doc/lifetimes.rst

```diff
@@ -0,0 +1,40 @@
+:man_page: bson_lifetimes
+
+:symbol:`bson_t` lifetimes
+==============
+A :symbol:`bson_t` may contain its data directly or may contain pointers to heap-allocated memory. Overwriting an existing :symbol:`bson_t`
+or allowing a stack-allocated :symbol:`bson_t` to go out of scope may cause a memory leak. A :symbol:`bson_t` should always be destroyed with
+:symbol:`bson_destroy()`.
+
+:symbol:`bson_t` out parameters
+-------------------
+A :symbol:`bson_t` pointer used as an out parameter must point to valid overwritable storage for a new :symbol:`bson_t`.
```

Comment:
I would recommend distinctly enumerating the conditions of "overwritable storage", which must be one of:

1. Uninitialized storage for a `bson_t`.
2. A `bson_t` object that has been zero-initialized.
3. A `bson_t` object that has been initialized via `BSON_INITIALIZER`.
4. A `bson_t` object that was _not_ created via `bson_new` *and* has been destroyed via `bson_destroy`.

There are other cases, but I think these are the most important ones. `BSON_INITIALIZER` is the most important case since it bridges the uncanny valley between "live bson_t" and "overwritable storage".